drivers/dp/fast_bitbang.h

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+
+#if defined(CONFIG_SOC_SERIES_NRF52X)
+#define CPU_CLOCK		64000000U
+#else
+#define CPU_CLOCK		CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC
+#endif
+
+#if defined(CONFIG_SOC_SERIES_NRF52X) \
+	|| defined(CONFIG_SOC_SERIES_NRF53X) \
+	|| defined(CONFIG_SOC_FAMILY_LPC)
+#define FAST_BITBANG_HW_SUPPORT 1
+#else
+#define FAST_BITBANG_HW_SUPPORT 0
+#endif
+
+static ALWAYS_INLINE void pin_delay_asm(uint32_t delay)
+{
+#if defined(CONFIG_CPU_CORTEX_M)
+	__asm volatile ("movs r3, %[p]\n"
+			".start_%=:\n"
+			"subs r3, #1\n"
+			"bne .start_%=\n"
+			:
+			: [p] "r" (delay)
+			: "r3", "cc"
+			);
+#endif
+}
+
+#if FAST_BITBANG_HW_SUPPORT
+
+static ALWAYS_INLINE void pin_platform_config_input(void *base, uint32_t port, uint8_t pin)
+{
+#if defined(CONFIG_SOC_SERIES_NRF52X) || defined(CONFIG_SOC_SERIES_NRF53X)
+	NRF_GPIO_Type * reg = base;
+
+	reg->PIN_CNF[pin] = 0b0000;
+#elif defined(CONFIG_SOC_FAMILY_LPC)
+	GPIO_Type *gpio_base = base;
+	gpio_base->DIRCLR[port] = BIT(pin);
+#endif
+}
+
+static ALWAYS_INLINE void pin_platform_config_output(void *base, uint32_t port, uint8_t pin)
+{
+#if defined(CONFIG_SOC_SERIES_NRF52X) || defined(CONFIG_SOC_SERIES_NRF53X)
+	NRF_GPIO_Type * reg = base;
+
+	reg->PIN_CNF[pin] = 0b0001;
+#elif defined(CONFIG_SOC_FAMILY_LPC)
+	GPIO_Type *gpio_base = base;
+
+	gpio_base->DIRSET[port] = BIT(pin);
+#endif
+}
+
+
+static ALWAYS_INLINE void pin_platform_set(void *base, uint32_t port, uint8_t pin)
+{
+#if defined(CONFIG_SOC_SERIES_NRF52X) || defined(CONFIG_SOC_SERIES_NRF53X)
+	NRF_GPIO_Type * reg = base;
+
+	reg->OUTSET = BIT(pin);
+#elif defined(CONFIG_SOC_FAMILY_LPC)
+	GPIO_Type *gpio_base = base;
+
+	gpio_base->SET[port] = BIT(pin);
```

Comment:
This is still wrong because gpio_base->SET[port] can be passed as base, there is no reason for port parameter.

Reply:
Currently, `SW_GPIOS_GET_REG` is used to get the register, which points to `gpio_base`. If we wanted to pass the `SET[port]` register directly, we would have to do chip-specific things in `SW_DEVICE_DEFINE`. I would like to avoid that.

Reply:
Also, SET[..] is not the only register, we also have a DIRSET[..] and DIRCLR[..] register..

Reply:
Yes and we do not need port parameter for that, please drop this "LPC" patch and the relevant changes in your patches.

Reply:
done